src/index.js

```diff
@@ -111,17 +111,30 @@ class EventsCoreAPI {
    * =========================================================================
    */
 
+  /**
+   * @typedef {object} ProviderOptions
+   * @property {string} [providerMetadataId] Fetch by providerMetadataId for the consumer org
+   * @property {string} [instanceId] For Self registered providers, instanceId is a must while fetching by providerMetadataId
+   * @property {Array} [providerMetadataIds] Fetch all providers ( and all instances ) for the list of provider metadata ids
```

Comment:
I am not sure how it is normally done in JsDoc, but if the fields cannot coexists, can we have a union type instead?